### PR TITLE
stbt.pylint_plugin: Support pylint 2 / astroid 2

### PR DESCRIPTION
**Pull request by drothlis**

### CODE REVIEW

**Comment by drothlis** [comment]
P.S. This is necessary to support Python 3.7, where pylint 1.8.3 / astroid 1.6.0 fail with `RuntimeError: generator 'infer_name' raised StopIteration`.

File: stbt/pylint_plugin.py [hunk 1]
```diff
@@ -283,10 +283,8 @@ def _in_git_repo():
 def _infer(node):
     try:
         for inferred in node.infer():
-            # Sometimes when `infer()` fails it returns `YES` which returns
-            # True to everything, including `callable()` and
-            # `isinstance(YES, <anything else>)`, so it isn't useful.
-            if inferred == YES:
+            # When `infer()` fails it returns this singleton:
+            if inferred is Uninferable:
```
Comment (wmanley) [inline stbt/pylint_plugin.py:287]: Does this change work with pylint 1.8.3 too?  I guess that would imply `YES == Uninferable`.  Maybe `Uninferable` is like `ANY` and anything `==` `Uninferable`?

Comment (drothlis) [reply stbt/pylint_plugin.py:287]:
```
$ ipython
In [1]: import astroid

In [2]: astroid.__version__
Out[2]: '1.6.0'

In [3]: astroid.YES
Out[3]: Uninferable
```

Comment (wmanley) [reply stbt/pylint_plugin.py:287]: Hah, I see, even simpler.

Comment (drothlis) [reply stbt/pylint_plugin.py:287]: Yes, it works with pylint 1.8.3, which is still tested in our CI jobs "ubuntu1804-python2" and "ubuntu1804-python3".

**Pull request merged**